libs/components/colorpicker/testing/src/colorpicker/colorpicker-dropdown-harness.ts

```
@@ -0,0 +1,153 @@
+
+/**
+ * Harness for interacting with colorpicker dropdown in tests.
+ */
+export class SkyColorpickerDropdownHarness extends SkyComponentHarness {
+  /**
+   * @internal
+   */
+  public static hostSelector = '.sky-colorpicker-container';
+
+  #getApplyButton = this.locatorFor('.sky-btn-colorpicker-apply');
+  #getCancelButton = this.locatorFor('.sky-btn-colorpicker-close');
+  #getInputs = this.locatorForAll('input.sky-form-control');
+  #getSwatchesDefault = this.locatorForAll('button.sky-preset-color');
+  #getSwatchesModern = this.locatorForAll('button.sky-btn-link');
+
+  public static with(
+    filters: SkyColorpickerDropdownHarnessFilters,
+  ): HarnessPredicate<SkyColorpickerDropdownHarness> {
+    return SkyColorpickerDropdownHarness.getDataSkyIdPredicate(
+      filters,
+    ).addOption('id', filters.id, async (harness, id) =>
+      HarnessPredicate.stringMatches(await harness.getId(), id),
+    );
+  }
+
+  /**
+   * Clicks the colorpicker dropdown apply button.
+   */
+  public async clickApplyButton(): Promise<void> {
+    return (await this.#getApplyButton()).click();
+  }
+
+  /**
+   * Clicks the colorpicker dropdown cancel button.
+   */
+  public async clickCancelButton(): Promise<void> {
+    return (await this.#getCancelButton()).click();
+  }
+
+  /**
+   * Clicks a specified swatch in the color preset section.
+   * @param swatchHex Hex code of the swatch to click.
+   */
+  public async clickSwatch(swatchHex: string): Promise<void> {
+    const swatches = await this.getSwatchButtons();
+    const ariaLabel = `Preset Color: ${swatchHex}`;
+
+    for (const swatch of swatches) {
+      if ((await swatch.getAttribute('aria-label')) === ariaLabel) {
+        return await swatch.click();
+      }
+    }
+  }
+
+  /**
+   * Enters a value into the alpha input box.
+   * @param value A decimal value from 0-1.
+   */
+  public async setAlphaValue(value: string): Promise<void> {
+    const input = await this.getAlphaInput();
```

This should throw a helpful error if the alpha button cannot be found (since it's enabled by the `allowTransparency` flag).